python/ray/dag/tests/test_accelerated_dag.py

```diff
@@ -176,6 +192,32 @@ def f(x):
         dag.experimental_compile()
 
 
+@pytest.mark.parametrize("num_actors", [1, 4])
+def test_dag_fault_tolerance(ray_start_regular, num_actors):
+    actors = [Actor.remote(0, fail_after=100) for _ in range(num_actors)]
+    with InputNode() as i:
+        out = [a.inc.bind(i) for a in actors]
+        dag = MultiOutputNode(out)
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(99):
+        output_channels = compiled_dag.execute(1)
+        # TODO(swang): Replace with fake ObjectRef.
+        results = [chan.begin_read() for chan in output_channels]
+        assert results == [i + 1] * num_actors
+        for chan in output_channels:
+            chan.end_read()
+
+    with pytest.raises(ValueError):
+        for i in range(99):
+            output_channels = compiled_dag.execute(1)
+            for chan in output_channels:
+                chan.begin_read()
+            for chan in output_channels:
+                chan.end_read()
```

Comment:
Q: is end_read necessary to call btw? 

Since now begin_read can raise an exception, it is easy to forget calling end_read.